### DISCUSSION TITLE
Android Studio - Data Lossing when the application restarts/moving to another activity.

### DISCUSSION THREAD

**Discussion by santhoshsivan12** [romannurik/AndroidAssetStudio#255] [Q&A]
I have a trouble! Please anyone help on this! I'm currently doing a food application for my college! Now the - problem is when ever I click the add button! 
**SAMPLE: **
[Screenshot_2022-08-16-21-41-36-11_5bc418c57209d4c065228a8244aa2e83.jpg](https://user-images.githubusercontent.com/84382634/184927770-7fac7950-9579-4e36-8bb9-f4cc292dace3.jpg)

Logically it's just working fine! But when ever I go to another activity or cart activity and came back to the the home activity it's not showing the current status of the food item! 

**Before go to another activity: **
[Screenshot_2022-08-16-21-44-24-93_5bc418c57209d4c065228a8244aa2e83.jpg](https://user-images.githubusercontent.com/84382634/184928302-7d4bdbd3-0bb8-4ffb-ae91-2bbf254e850d.jpg)

**After gone to another activity: **
![Screenshot_2022-08-16-21-41-36-11_5bc418c57209d4c065228a8244aa2e83.jpg](https://user-images.githubusercontent.com/84382634/184928734-befe51cc-7d7e-4e26-8a76-79a81eee2820.jpg)

Please somebody help me! Thanks in advance :)

**Comment by papaBearClinton** [comment]
If the activity is restarting then all states will be set to its default  just like what you would want when the activity starts for the first time.

One way to save the state is to well save the state in the onSaveInstanceState method of the activity.

In your onCreate method you can get what you saved which in turn you can use to restore your state

Warning --this just save the state while the app is running or in the background. If the app is killed and then restarted all states will be set to its default 

my advice will be to save the state of the app on a server since it's a delivery app and you will want to have a copy of what the user ordered
